.github/workflows/cli-ci.yaml

```diff
@@ -107,11 +116,36 @@ jobs:
   build_test:
     runs-on: ubuntu-latest
     needs: [build_test_default, build_test_node_matrix, build_bun_matrix]
-    if: always()
+    if: github.event.action != 'closed' && always()
     steps:
       - name: 'All tests passed'
         if: ${{ !(contains(needs.*.result, 'failure')) }}
         run: exit 0
       - name: 'Some tests failed'
         if: ${{ contains(needs.*.result, 'failure') }}
         run: exit 1
+
+  post-merge:
+    if: >
+      github.event_name == 'pull_request' &&
+      github.event.action == 'closed' &&
+      github.event.pull_request.merged == false
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: 2be5a84f75694cc6e7a99960be2e8e45b2cd3999
```

Comment:
Added for testing to see if we can upload on close (without merging), will revert before merging pull request